pkg/cli/image_cmd_test.go

```diff
@@ -1552,42 +1552,35 @@ func uploadTestMultiarch(baseURL string) {
 	layer11 := []byte{11, 12, 13, 14}
 	layer12 := []byte{16, 17, 18, 19}
 
-	image1, err := test.GetImageWithComponents(
-		ispec.Image{
-			Platform: ispec.Platform{
-				OS:           "linux",
-				Architecture: "amd64",
-			},
-		},
-		[][]byte{
+	image1 := test.CreateImageWith().
+		LayerBlobs([][]byte{
 			layer11,
 			layer12,
-		},
-	)
-	So(err, ShouldBeNil)
+		}).
+		ImageConfig(
+			ispec.Image{
+				Platform: ispec.Platform{OS: "linux", Architecture: "amd64"},
+			},
+		).Build()
 
 	// ------ Define Image2
 	layer21 := []byte{21, 22, 23, 24}
 
-	image2, err := test.GetImageWithComponents(
-		ispec.Image{
-			Platform: ispec.Platform{
-				OS:           "windows",
-				Architecture: "arm64",
-				Variant:      "v6",
-			},
-		},
-		[][]byte{
+	image2 := test.CreateImageWith().
+		LayerBlobs([][]byte{
 			layer21,
-		},
-	)
-	So(err, ShouldBeNil)
+		}).
+		ImageConfig(
+			ispec.Image{
+				Platform: ispec.Platform{OS: "windows", Architecture: "arm64", Variant: "v6"},
+			},
+		).Build()
 
 	// ------- Upload The multiarch image
 
 	multiarch := test.GetMultiarchImageForImages("multi-arch", []test.Image{image1, image2})
 
-	err = test.UploadMultiarchImage(multiarch, baseURL, "repo")
+	err := test.UploadMultiarchImage(multiarch, baseURL, "repo")
```

Comment:
Remove the tag handling completely from the image creation.
Specify the tag in the upload function parameters.